lib/rouge/lexers/hocon.rb

```diff
@@ -11,33 +11,59 @@ class HOCON < JSON
       tag 'hocon'
       filenames '*.hocon'
 
-      prepend :root do
+      state :comments do
         # Comments
         rule %r(//.*?$), Comment::Single
         rule %r(#.*?$), Comment::Single
+      end
+
+      prepend :root do
+        mixin :comments
+      end
+
+      prepend :object do
+        # Keywords
+        rule %r/\b(?:include|url|file|classpath)\b/, Keyword
+      end
+
+      state :name do
+        rule %r/("(?:\"|[^"\n])*?")(\s*)([:=]|(?={))/ do
+          groups Name::Label, Text::Whitespace, Punctuation
+        end
+
+        rule %r/([-\w.]+)(\s*)([:=]|(?={))/ do
+          groups Name::Label, Text::Whitespace, Punctuation
+        end
+      end
+
+      state :value do
+        mixin :comments
+
+        rule %r/\n/, Text::Whitespace
+        rule %r/\s+/, Text::Whitespace
+
+        mixin :constants
 
         # Interpolation
         rule %r/[$][{][?]?/, Literal::String::Interpol, :interpolation
 
         # Strings
         rule %r/"""/, Literal::String::Double, :multiline_string
+        rule %r/"/, Str::Double, :string
 
-        # Keywords
-        rule %r/\b(?:include|url|file|classpath)\b/, Keyword
+        rule %r/\[/, Punctuation, :array
+        rule %r/{/, Punctuation, :object
```

Comment:
can't these be replaced by the namesakes from the JSON superclass?

Reply:
@ashmaroli Not sure what you mean here. The references are interpreted as references to `:array` and `:object` as defined in the JSON lexer.

Reply:
Oh.. I thought we could just `mixin :array` here.. Guess, I hoped wrong then.